src/Compilers/CSharp/Portable/Binder/Binder.NamespaceOrTypeOrAliasSymbolWithAnnotations.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal partial class Binder
+    {
+        internal struct NamespaceOrTypeOrAliasSymbolWithAnnotations
+        {
+            private readonly object _symbolOrTypeSymbolWithAnnotations;
+
+            private NamespaceOrTypeOrAliasSymbolWithAnnotations(object symbolOrTypeSymbolWithAnnotations)
+            {
+                Debug.Assert(symbolOrTypeSymbolWithAnnotations != null);
+                Debug.Assert(!(symbolOrTypeSymbolWithAnnotations is TypeSymbol));
+                _symbolOrTypeSymbolWithAnnotations = symbolOrTypeSymbolWithAnnotations;
+            }
+
+            internal TypeSymbolWithAnnotations Type => _symbolOrTypeSymbolWithAnnotations as TypeSymbolWithAnnotations;
+            internal Symbol Symbol => _symbolOrTypeSymbolWithAnnotations as Symbol;
+            internal bool IsType => !(Type is null);
+            internal bool IsAlias => Symbol?.Kind == SymbolKind.Alias;
+            internal Symbol SymbolOrType => Symbol ?? Type?.TypeSymbol;
+            internal NamespaceOrTypeSymbol NamespaceOrTypeSymbol => (NamespaceOrTypeSymbol)SymbolOrType;
+            internal bool IsDefault => Symbol is null && Type is null;
```

Comment:
Would it be sufficient to test `_symbolOrTypeSymbolWithAnnotations == null` for `IsDefault`?